compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -97,6 +97,20 @@ impl<'tcx, Tag: Provenance> std::ops::Deref for PlaceTy<'tcx, Tag> {
     }
 }
 
+pub enum ImmediateOrMPlace<'tcx, Tag: Provenance> {
+    Imm(ImmTy<'tcx, Tag>),
+    MPlace(MPlaceTy<'tcx, Tag>),
+}
+
+impl<'tcx, Tag: Provenance> ImmediateOrMPlace<'tcx, Tag> {
+    pub fn get_mplace(self) -> MPlaceTy<'tcx, Tag> {
```

Comment:
Maybe `unwrap_mplace` would be better? At least it indicates that this might panic.